server/src/main/java/org/jetbrains/bsp/bazel/server/bsp/services/JvmBuildServerService.java

```diff
@@ -58,8 +69,18 @@ public JvmBuildServerService(BazelData bazelData, BazelRunner bazelRunner) {
   }
 
   private List<JvmEnvironmentItem> getJvmEnvironmentItems(List<BuildTargetIdentifier> targets) {
-    return targetsLanguageOptionsResolver.getResultItemsForTargets(targets).stream()
-        .distinct()
-        .collect(Collectors.toList());
+    var items =
+        targetsLanguageOptionsResolver.getResultItemsForTargets(targets).stream()
+            .distinct()
+            .collect(Collectors.toList());
+    items.forEach(item -> item.setClasspath(fetchClasspath(item.getTarget())));
+    return items;
```

Comment:
```suggestion
    return targetsLanguageOptionsResolver.getResultItemsForTargets(targets).stream()
            .distinct()
            .peek(item -> item.setClasspath(fetchClasspath(item.getTarget())))
            .collect(Collectors.toList());
```

Reply:
I did that originally, but this is actually modifying the objects in the collection at some unpredicted time. This means that it affects the calculation of disticts and yields incorrect results.

Reply:
right, stream things